sound/soc/sof/intel/hda.c

```diff
@@ -406,9 +406,20 @@ static irqreturn_t hda_dsp_interrupt_handler(int irq, void *context)
 {
 	struct snd_sof_dev *sdev = context;
 
-	if (hda_dsp_check_ipc_irq(sdev) ||
-	    hda_dsp_check_stream_irq(sdev))
+	/* clear flags for interrupt sources */
+	sdev->irq_event = 0;
+
+	if (hda_dsp_check_stream_irq(sdev) ||
+	    hda_dsp_check_ipc_irq(sdev)) {
```

Comment:
Just note that ``hda_dsp_check_ipc_irq()`` will not be called if ``hda_dsp_check_stream_irq()`` return true. However  ``hda_dsp_check_ipc_irq()`` will be called in ``hda_dsp_interrupt_thread()`` in that case. So, indeed, we don't need to check remaining event in the handler when there is already one event is confirmed.